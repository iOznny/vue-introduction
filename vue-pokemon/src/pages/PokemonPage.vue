<template>
    <div>
        <h1>Â¿Quien es este Pokemon?</h1>

        <!-- Picture -->
        <PokemonPicture v-bind:pokemonID="1" :showPokemon="true" />

        <!-- Options -->
        <PokemonOptions />

    </div>
</template>

<script>
import PokemonPicture from '@/components/PokemonPicture.vue';
import PokemonOptions from '@/components/PokemonOptions.vue';

import getPokemons from '@/helpers/getPokemosOptions';

export default {
    name: 'PokemonPage',
    components: {
        PokemonPicture,
        PokemonOptions,
    }
}
</script>