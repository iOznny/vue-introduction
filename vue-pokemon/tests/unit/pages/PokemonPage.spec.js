import { shallowMount, mount } from '@vue/test-utils';
import { mockPokemons } from '../mocks/pokemons.mock';

import PokemonPage from '@/pages/PokemonPage';

describe('Page - Pokemon Page', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage);
    })
    
    test('Match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Call mixPokemons', () => {
        const mixPokemonsSpy = jest.spyOn(PokemonPage.methods, 'mixPokemons');
        const wrapper = shallowMount(PokemonPage);

        expect(mixPokemonsSpy).toHaveBeenCalled();
    })

    test('Match with snapshot when loading pokemons', () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemons: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Show components PokemonPicture and PokemonOptions', () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemons: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        });
        
        const picture = wrapper.find('pokemon-picture-stub')
        const options = wrapper.find('pokemon-options-stub')

        // Components exists
        expect(picture.exists()).toBeTruthy();
        expect(options.exists()).toBeTruthy();

        // PokemonPicture has attribute id === 5
        expect(picture.attributes('pokemonid')).toBe('1');

        // PokemonOptions had attribute pokemons to be true
        expect(options.attributes('pokemons')).toBeTruthy();
    })

    test('Testing with checkAnswer', async () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemons: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        });

        await wrapper.vm.checkAnswer(1);

        expect(wrapper.find('h2').exists()).toBeTruthy();
        expect(wrapper.vm.showPokemon).toBeTruthy();

        await wrapper.vm.checkAnswer(1);
        expect(wrapper.vm.message).toBe('Â¡Correcto! PIKACHU');
    })
})