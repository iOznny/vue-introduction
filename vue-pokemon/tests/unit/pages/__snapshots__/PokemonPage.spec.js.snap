// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - Pokemon Page Match with snapshot 1`] = `
<div>
  <div>
    <h1>Espere por favor...</h1>
  </div>
</div>
`;

exports[`Page - Pokemon Page Match with snapshot when loading pokemons 1`] = `
<div>
  <div>
    <h1>Â¿Quien es este Pokemon?</h1><!-- Picture -->
    <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub><!-- Options -->
    <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
    <!--v-if-->
  </div>
</div>
`;
